protocols/v1/src/lib.rs

```diff
@@ -443,16 +452,23 @@ pub trait IsClient {
                     // Err(())
                     panic!("TODO: Check if version_bits is set");
                 } else if self.is_authorized(&user_name) {
-                    Ok(client_to_server::Submit {
-                        job_id: self.last_notify().unwrap().job_id,
-                        user_name,
-                        extra_nonce2,
-                        time: HexU32Be(time as u32),
-                        nonce: HexU32Be(nonce as u32),
-                        version_bits,
-                        id,
+                    if let Some(notify) = self.last_notify() {
+                        Ok(client_to_server::Submit {
+                            job_id: notify.job_id,
+                            user_name,
+                            extra_nonce2,
+                            time: HexU32Be(time as u32),
+                            nonce: HexU32Be(nonce as u32),
+                            version_bits,
+                            id,
+                        }
+                        .into())
+                    } else {
+                        // TODO: create
```

Comment:
Isn't this case impossible because the if case above checks `if self.last_notify().is_none(){panic!();}`